python/tests/conftest.py

```diff
@@ -278,6 +305,7 @@ def mongo_version_store(mongo_store_factory):
     params=[
         "s3_store_factory",
         pytest.param("azure_store_factory", marks=AZURE_TESTS_MARK),
+        pytest.param("azure_no_ssl_store_factory", marks=AZURE_TESTS_MARK),
```

Comment:
This is unclean, I don't think we should be adding so many more Azure test runs without a good reason. Surely we don't need to run every Azure test both with and without SSL?